### ISSUE TITLE
[JENKINS-67984] Jenkins Git plugin only checks out tags on existing repository

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#3714]
<p>When specifying tag checkout behaviour explicitly in the git plugin, tags are only checked out if the workspace already exists and contains a git repository.</p>

<p>Reproduce by creating a git repository containing the following <tt>Jenkinsfile</tt>:</p>

<pre>
pipeline {
    agent any
    options { skipDefaultCheckout(<span class="code-keyword">true</span>) }
    triggers { githubPush() }
    stages {
stage(<span class="code-quote">'Clone repository'</span>) {
    steps {
cleanWs()
checkout([
    $class: <span class="code-quote">'GitSCM'</span>,
    branches: scm.branches,
    doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
    extensions: scm.extensions + [[$class: <span class="code-quote">'CloneOption'</span>, noTags: <span class="code-keyword">false</span>, reference: '', shallow: <span class="code-keyword">false</span>]],
    submoduleCfg: [],
    userRemoteConfigs: scm.userRemoteConfigs
])
checkout([
    $class: <span class="code-quote">'GitSCM'</span>,
    branches: scm.branches,
    doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
    extensions: scm.extensions + [[$class: <span class="code-quote">'CloneOption'</span>, noTags: <span class="code-keyword">false</span>, reference: '', shallow: <span class="code-keyword">false</span>]],
    submoduleCfg: [],
    userRemoteConfigs: scm.userRemoteConfigs
])
    }
}
    }
}
</pre>
<p> and you will see this happens in the build log - I have attached the full build log for reference:</p>

```
10:47:01   > /usr/bin/git fetch --no-tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
10:47:02   > /usr/bin/git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10

```

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilcatania">ilcatania</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984" target="_blank">Jenkins Git plugin only checks out tags on existing repository</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: git-plugin
<li><b>label(s)</b>: git
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When specifying tag checkout behaviour explicitly in the git plugin, tags are only checked out if the workspace already exists and contains a git repository.</p>

<p>Reproduce by creating a git repository containing the following <tt>Jenkinsfile</tt>:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">pipeline {
    agent any
    options { skipDefaultCheckout(<span class="code-keyword">true</span>) }
    triggers { githubPush() }
    stages {
        stage(<span class="code-quote">'Clone repository'</span>) {
            steps {
                cleanWs()
                checkout([
                    $class: <span class="code-quote">'GitSCM'</span>,
                    branches: scm.branches,
                    doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
                    extensions: scm.extensions + [[$class: <span class="code-quote">'CloneOption'</span>, noTags: <span class="code-keyword">false</span>, reference: '', shallow: <span class="code-keyword">false</span>]],
                    submoduleCfg: [],
                    userRemoteConfigs: scm.userRemoteConfigs
                ])
                checkout([
                    $class: <span class="code-quote">'GitSCM'</span>,
                    branches: scm.branches,
                    doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
                    extensions: scm.extensions + [[$class: <span class="code-quote">'CloneOption'</span>, noTags: <span class="code-keyword">false</span>, reference: '', shallow: <span class="code-keyword">false</span>]],
                    submoduleCfg: [],
                    userRemoteConfigs: scm.userRemoteConfigs
                ])
            }
        }
    }
}
</pre>
</div></div>
<p> and you will see this happens in the build log - I have attached the full build log for reference:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>10:47:01   &gt; /usr/bin/git fetch --no-tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
10:47:02   &gt; /usr/bin/git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
</pre>
</div></div>
</pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 2.319.3<br/>
org.jenkins-ci.main:jenkins-war:2.319.3<br/>
org.jenkins-ci.plugins:git-client:3.11.0<br/>
org.jenkins-ci.plugins:git:4.10.3<br/>
git version 2.35
```
</details>
<details><summary><i>1 attachment</i></summary>

- [jenkins-no-git-full-log.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/57418/jenkins-no-git-full-log.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=422862&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-422862">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>To clarify, I've marked the severity as major because several versioning schemes use existing tags to calculate CI/CD build versions, and they calculate the wrong version if no tags are checked out.
It took me several hours to troubleshoot this issue after noticing my build artifact versions were calculated incorrectly. I would imagine others will be similarly impacted.
For examples of versioning tools broken by this bug: <a href="https://jgitver.github.io/" class="external-link" target="_blank" rel="nofollow noopener">jgitver</a> (maven/gradle), <a href="https://github.com/pypa/setuptools_scm/" class="external-link" target="_blank" rel="nofollow noopener">setuptools_scm</a> (python), <a href="https://index.scala-lang.org/rallyhealth/sbt-git-versioning/sbt-git-versioning/" class="external-link" target="_blank" rel="nofollow noopener">sbt-git-versioning</a> (scala)</p>
</pre>
</details>

</ul>
</details>

<p>To clarify, I've marked the severity as major because several versioning schemes use existing tags to calculate CI/CD build versions, and they calculate the wrong version if no tags are checked out.<br/>
It took me several hours to troubleshoot this issue after noticing my build artifact versions were calculated incorrectly. I would imagine others will be similarly impacted.<br/>
For examples of versioning tools broken by this bug: <a href="https://jgitver.github.io/" class="external-link" target="_blank" rel="nofollow noopener">jgitver</a> (maven/gradle), <a href="https://github.com/pypa/setuptools_scm/" class="external-link" target="_blank" rel="nofollow noopener">setuptools_scm</a> (python), <a href="https://index.scala-lang.org/rallyhealth/sbt-git-versioning/sbt-git-versioning/" class="external-link" target="_blank" rel="nofollow noopener">sbt-git-versioning</a> (scala)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=423121&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423121">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> can you please confirm you're able to reproduce the issue? Do you have any pointers on how to fix please?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> can you please confirm you're able to reproduce the issue? Do you have any pointers on how to fix please?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=423123&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423123">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilcatania" class="user-hover" rel="ilcatania">ilcatania</a> I am out of the office spending time with family.  I won't investigate this until the middle of next week at the earliest.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilcatania" class="user-hover" rel="ilcatania">ilcatania</a> I am out of the office spending time with family.  I won't investigate this until the middle of next week at the earliest.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=423712&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423712">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ok thank you for replying back. If any more info is needed or this should be assigned to someone else please let me know.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ok thank you for replying back. If any more info is needed or this should be assigned to someone else please let me know.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=424079&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424079">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi, any news on this one please? Would appreciate pointers on how to solve, thank you very much</p>
</pre>
</details>

</ul>
</details>

<p>Hi, any news on this one please? Would appreciate pointers on how to solve, thank you very much</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=424318&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424318">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> can you help with this? Or can you point me to someone who can please? Thanks</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> can you help with this? Or can you point me to someone who can please? Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=424328&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424328">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilcatania" class="user-hover" rel="ilcatania">ilcatania</a> but the time that I've been able to give to the git plugin and specific questions like this has been limited by my involvement in other projects like <a href="https://www.jenkins.io/blog/2022/04/11/she-code-africa-contributhon/" class="external-link" target="_blank" rel="nofollow noopener">She Code Africa Contributhon</a> and <a href="https://www.jenkins.io/projects/gsoc/" class="external-link" target="_blank" rel="nofollow noopener">Google Summer of Code</a>.  It will likely be several more weeks before I could look into this further.</p>

<p>You could ask the question on community.jenkins.io to see if others there could help.</p>

<p>You could ask the question in the gitter chat channels in case someone offers help there.</p>
</pre>
</details>

</ul>
</details>

<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ilcatania" class="user-hover" rel="ilcatania">ilcatania</a> but the time that I've been able to give to the git plugin and specific questions like this has been limited by my involvement in other projects like <a href="https://www.jenkins.io/blog/2022/04/11/she-code-africa-contributhon/" class="external-link" target="_blank" rel="nofollow noopener">She Code Africa Contributhon</a> and <a href="https://www.jenkins.io/projects/gsoc/" class="external-link" target="_blank" rel="nofollow noopener">Google Summer of Code</a>.  It will likely be several more weeks before I could look into this further.</p>

<p>You could ask the question on community.jenkins.io to see if others there could help.</p>

<p>You could ask the question in the gitter chat channels in case someone offers help there.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425161&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425161">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've tried posting this in the jenkins community forum here: <a href="https://community.jenkins.io/t/jenkins-git-plugin-only-checks-out-tags-on-existing-repository/2101" class="external-link" target="_blank" rel="nofollow noopener">https://community.jenkins.io/t/jenkins-git-plugin-only-checks-out-tags-on-existing-repository/2101</a> but the only reply I got was from a person telling me I was doing things wrong, and pointing me to the git plugin source code.</p>

<p>I've also posted in the gitter channels (which I can't easily link to from my work computer since infosec blocks the URLs) but have not gotten a single answer there.</p>

<p>It would be great if someone could at least independently confirm that this is a bug, I spent quite some time trying to build a self contained reproducible example and I am a bit frustrated that I have not gotten at least some kind of feedback on whether the issue is legitimate or not.</p>
</pre>
</details>

</ul>
</details>

<p>I've tried posting this in the jenkins community forum here: <a href="https://community.jenkins.io/t/jenkins-git-plugin-only-checks-out-tags-on-existing-repository/2101" class="external-link" target="_blank" rel="nofollow noopener">https://community.jenkins.io/t/jenkins-git-plugin-only-checks-out-tags-on-existing-repository/2101</a> but the only reply I got was from a person telling me I was doing things wrong, and pointing me to the git plugin source code.</p>

<p>I've also posted in the gitter channels (which I can't easily link to from my work computer since infosec blocks the URLs) but have not gotten a single answer there.</p>

<p>It would be great if someone could at least independently confirm that this is a bug, I spent quite some time trying to build a self contained reproducible example and I am a bit frustrated that I have not gotten at least some kind of feedback on whether the issue is legitimate or not.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/saper.png" title="saper's avatar" /> saper:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425170&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425170">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a very good issue report, thank you. Didn't have time to test and reproduce but:</p>

<p>Can you check if enabling <b>Preserve second fetch during checkout</b> in the global configuration helps?</p>

<p>I can see that there is an initial fetch being done <a href="https://github.com/jenkinsci/git-plugin/blob/git-4.11.1/src/main/java/hudson/plugins/git/GitSCM.java#L1212-L1221" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/git-4.11.1/src/main/java/hudson/plugins/git/GitSCM.java#L1212-L1221</a> without regard if tags are needed or not. After this, <tt>determineSecondFetch</tt> is being run and I think there is no regard whether tags are needed or not, either.</p>

<p>Finally, the plugin decided to <tt>10:47:01  Avoid second fetch</tt> as seen in your log.</p>

<p>There are some tests for this like <a href="https://github.com/jenkinsci/git-plugin/blob/8d93ac3e1141c3e3f645569be9cfaa0cc9d61b62/src/test/java/hudson/plugins/git/GitSCMTest.java#L555-L583" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/8d93ac3e1141c3e3f645569be9cfaa0cc9d61b62/src/test/java/hudson/plugins/git/GitSCMTest.java#L555-L583</a></p>

<p>I have to admit I never understood the fetch logic of this plugin and it seems convoluted indeed.</p>
</pre>
</details>

</ul>
</details>

<p>This is a very good issue report, thank you. Didn't have time to test and reproduce but:</p>

<p>Can you check if enabling <b>Preserve second fetch during checkout</b> in the global configuration helps?</p>

<p>I can see that there is an initial fetch being done <a href="https://github.com/jenkinsci/git-plugin/blob/git-4.11.1/src/main/java/hudson/plugins/git/GitSCM.java#L1212-L1221" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/git-4.11.1/src/main/java/hudson/plugins/git/GitSCM.java#L1212-L1221</a> without regard if tags are needed or not. After this, <tt>determineSecondFetch</tt> is being run and I think there is no regard whether tags are needed or not, either.</p>

<p>Finally, the plugin decided to <tt>10:47:01  Avoid second fetch</tt> as seen in your log.</p>

<p>There are some tests for this like <a href="https://github.com/jenkinsci/git-plugin/blob/8d93ac3e1141c3e3f645569be9cfaa0cc9d61b62/src/test/java/hudson/plugins/git/GitSCMTest.java#L555-L583" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/8d93ac3e1141c3e3f645569be9cfaa0cc9d61b62/src/test/java/hudson/plugins/git/GitSCMTest.java#L555-L583</a></p>

<p>I have to admit I never understood the fetch logic of this plugin and it seems convoluted indeed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/saper.png" title="saper's avatar" /> saper:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425173&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425173">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Strange, I could reproduce this only once on a fresh Jenkins installation.</p>

<p>Now I always see tags being fetched:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">Obtained Jenkinsfile from git https://github.com/ilCatania/jenkins-git-no-tags
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /home/saper/.jenkins/workspace/original-no-tags
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Clone repository)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git init /home/saper/.jenkins/workspace/original-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git --version # timeout=10
 &gt; git --version # 'git version 2.36.0'
 &gt; git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 &gt; git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
 &gt; git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 &gt; git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
First time build. Skipping changelog.
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
 &gt; git rev-parse --resolve-git-dir /home/saper/.jenkins/workspace/original-no-tags/.git # timeout=10
Fetching changes from the remote Git repository
 &gt; git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git --version # timeout=10
 &gt; git --version # 'git version 2.36.0'
 &gt; git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 &gt; git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS </pre>
</div></div>

<p>Tried to reproduce with</p>

<p>Jenkins 2.332.2
Git client plugin Version 3.11.0 
 Git plugin Version 4.11.1 </p>
</pre>
</details>

</ul>
</details>

<p>Strange, I could reproduce this only once on a fresh Jenkins installation.</p>

<p>Now I always see tags being fetched:</p>


<pre>
Obtained Jenkinsfile from git https://github.com/ilCatania/jenkins-git-no-tags
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /home/saper/.jenkins/workspace/original-no-tags
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Clone repository)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/ilCatania/jenkins-git-no-tags
 > git init /home/saper/.jenkins/workspace/original-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 > git --version # timeout=10
 > git --version # 'git version 2.36.0'
 > git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
First time build. Skipping changelog.
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /home/saper/.jenkins/workspace/original-no-tags/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 > git --version # timeout=10
 > git --version # 'git version 2.36.0'
 > git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS </pre>

<p>Tried to reproduce with</p>

<p>Jenkins 2.332.2<br/>
Git client plugin Version 3.11.0 <br/>
 Git plugin Version 4.11.1 </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/saper.png" title="saper's avatar" /> saper:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425182&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425182">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have fixed some other issues with this Jenkins (plugin installation) and I cannot reproduce it:</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">Obtained Jenkinsfile from git https://github.com/ilCatania/jenkins-git-no-tags
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /home/saper/.jenkins/workspace/test-git
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Clone repository)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git init /home/saper/.jenkins/workspace/test-git # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git --version # timeout=10
 &gt; git --version # 'git version 2.36.0'
 &gt; git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 &gt; git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
 &gt; git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 &gt; git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
First time build. Skipping changelog.
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
 &gt; git rev-parse --resolve-git-dir /home/saper/.jenkins/workspace/test-git/.git # timeout=10
Fetching changes from the remote Git repository
 &gt; git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 &gt; git --version # timeout=10
 &gt; git --version # 'git version 2.36.0'
 &gt; git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 &gt; git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
 </pre>
</div></div>

<p>I've had some issues with locale settings, are you running Jenkins with <tt>LANG=C</tt> ?</p>
</pre>
</details>

</ul>
</details>

<p>I have fixed some other issues with this Jenkins (plugin installation) and I cannot reproduce it:</p>

<p> </p>

<pre>
Obtained Jenkinsfile from git https://github.com/ilCatania/jenkins-git-no-tags
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /home/saper/.jenkins/workspace/test-git
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Clone repository)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/ilCatania/jenkins-git-no-tags
 > git init /home/saper/.jenkins/workspace/test-git # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 > git --version # timeout=10
 > git --version # 'git version 2.36.0'
 > git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
First time build. Skipping changelog.
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /home/saper/.jenkins/workspace/test-git/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags # timeout=10
Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags
 > git --version # timeout=10
 > git --version # 'git version 2.36.0'
 > git fetch --tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0d6b4accd5eba0c2e8efaa47fac5dcb76b030201 # timeout=10
Commit message: "initial commit"
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
 </pre>

<p>I've had some issues with locale settings, are you running Jenkins with <tt>LANG=C</tt> ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilcatania.jpg" title="ilcatania's avatar" /> ilcatania:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425184&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425184">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=saper" class="user-hover" rel="saper">saper</a> thank you for looking into it.</p>

<p>I am running with <tt>LANG=en_US.UTF-8</tt> according to system information.</p>

<p>I tried enabling the "preserve second fetch" option, which ironically in the tooltip says "If you have to enable this due to a bug, please create a jira" <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>Anyway, indeed with the "preserve second fetch" option it looks like the second fetch indeed looks at tags, but to be honest now I can't quite tell for sure what's happening:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>[2022-04-27T20:26:39.452Z] [Pipeline] checkout
[2022-04-27T20:26:39.469Z] The recommended git tool is: NONE
[2022-04-27T20:26:39.637Z] using credential github-app
[2022-04-27T20:26:39.653Z] Cloning the remote Git repository
[2022-04-27T20:26:39.653Z] Cloning with configured refspecs honoured and without tags
[2022-04-27T20:26:40.096Z] Pruning obsolete local branches
[2022-04-27T20:26:40.096Z] Pruning obsolete local tags that do not exist on remotes
[2022-04-27T20:26:40.112Z] Fetching without tags
[2022-04-27T20:26:39.673Z] Cloning repository https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:39.673Z]  &gt; /export/home/anaconda/2020.02/bin/git init /export/data/ci/jenkins/workspace/s_jenkins-git-no-tags_master # timeout=10
[2022-04-27T20:26:39.682Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:39.682Z]  &gt; /export/home/anaconda/2020.02/bin/git --version # timeout=10
[2022-04-27T20:26:39.685Z]  &gt; git --version # 'git version 2.35.3'
[2022-04-27T20:26:39.685Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:39.686Z]  &gt; /export/home/anaconda/2020.02/bin/git fetch --no-tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.066Z]  &gt; /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:40.070Z]  &gt; /export/home/anaconda/2020.02/bin/git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.086Z]  &gt; /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:40.102Z]  &gt; /export/home/anaconda/2020.02/bin/git show-ref --tags -d # timeout=10
[2022-04-27T20:26:40.118Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:40.118Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:40.119Z]  &gt; /export/home/anaconda/2020.02/bin/git fetch --tags --force --progress --prune -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.666Z] Checking out Revision 7912da59315e5959de99fc58a190171d74f3198b (master)
[2022-04-27T20:26:40.706Z] Commit message: "retry double checkout"
[2022-04-27T20:26:40.672Z]  &gt; /export/home/anaconda/2020.02/bin/git config core.sparsecheckout # timeout=10
[2022-04-27T20:26:40.676Z]  &gt; /export/home/anaconda/2020.02/bin/git checkout -f 7912da59315e5959de99fc58a190171d74f3198b # timeout=10
[2022-04-27T20:26:40.712Z]  &gt; /export/home/anaconda/2020.02/bin/git rev-list --no-walk b065c52da9fb8e0a07ca2e0d3dba544438412bf6 # timeout=10
[2022-04-27T20:26:41.332Z] [GitHub Checks] GitHub check (name: Jenkins, status: in_progress) has been published.
[2022-04-27T20:26:42.008Z] [Pipeline] checkout
[2022-04-27T20:26:42.023Z] The recommended git tool is: NONE
[2022-04-27T20:26:42.193Z] using credential github-app
[2022-04-27T20:26:42.211Z] Fetching changes from the remote Git repository
[2022-04-27T20:26:42.229Z] Pruning obsolete local branches
[2022-04-27T20:26:42.229Z] Pruning obsolete local tags that do not exist on remotes
[2022-04-27T20:26:42.524Z] Fetching without tags
[2022-04-27T20:26:42.832Z] Checking out Revision 7912da59315e5959de99fc58a190171d74f3198b (master)
[2022-04-27T20:26:42.864Z] Commit message: "retry double checkout"
[2022-04-27T20:26:42.203Z]  &gt; /export/home/anaconda/2020.02/bin/git rev-parse --resolve-git-dir /export/data/ci/jenkins/workspace/s_jenkins-git-no-tags_master/.git # timeout=10
[2022-04-27T20:26:42.219Z]  &gt; /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:42.235Z]  &gt; /export/home/anaconda/2020.02/bin/git show-ref --tags -d # timeout=10
[2022-04-27T20:26:42.253Z]  &gt; /export/home/anaconda/2020.02/bin/git --version # timeout=10
[2022-04-27T20:26:42.256Z]  &gt; git --version # 'git version 2.35.3'
[2022-04-27T20:26:42.256Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:42.257Z]  &gt; /export/home/anaconda/2020.02/bin/git ls-remote -t -- https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:42.531Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:42.531Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:42.531Z]  &gt; /export/home/anaconda/2020.02/bin/git fetch --tags --force --progress --prune -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:42.839Z]  &gt; /export/home/anaconda/2020.02/bin/git config core.sparsecheckout # timeout=10
[2022-04-27T20:26:42.842Z]  &gt; /export/home/anaconda/2020.02/bin/git checkout -f 7912da59315e5959de99fc58a190171d74f3198b # timeout=10
[2022-04-27T20:26:43.545Z] [GitHub Checks] GitHub check (name: Jenkins, status: in_progress) has been published.
</pre>
</div></div>

<p>I guess my main point is, if someone is trying to set this up, he's in for a lot of weirdness and several hours trying to figure it out</p>
</pre>
</details>

</ul>
</details>

<p>hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=saper" class="user-hover" rel="saper">saper</a> thank you for looking into it.</p>

<p>I am running with <tt>LANG=en_US.UTF-8</tt> according to system information.</p>

<p>I tried enabling the "preserve second fetch" option, which ironically in the tooltip says "If you have to enable this due to a bug, please create a jira" <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>Anyway, indeed with the "preserve second fetch" option it looks like the second fetch indeed looks at tags, but to be honest now I can't quite tell for sure what's happening:</p>


```
[2022-04-27T20:26:39.452Z] [Pipeline] checkout
[2022-04-27T20:26:39.469Z] The recommended git tool is: NONE
[2022-04-27T20:26:39.637Z] using credential github-app
[2022-04-27T20:26:39.653Z] Cloning the remote Git repository
[2022-04-27T20:26:39.653Z] Cloning with configured refspecs honoured and without tags
[2022-04-27T20:26:40.096Z] Pruning obsolete local branches
[2022-04-27T20:26:40.096Z] Pruning obsolete local tags that do not exist on remotes
[2022-04-27T20:26:40.112Z] Fetching without tags
[2022-04-27T20:26:39.673Z] Cloning repository https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:39.673Z]  > /export/home/anaconda/2020.02/bin/git init /export/data/ci/jenkins/workspace/s_jenkins-git-no-tags_master # timeout=10
[2022-04-27T20:26:39.682Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:39.682Z]  > /export/home/anaconda/2020.02/bin/git --version # timeout=10
[2022-04-27T20:26:39.685Z]  > git --version # 'git version 2.35.3'
[2022-04-27T20:26:39.685Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:39.686Z]  > /export/home/anaconda/2020.02/bin/git fetch --no-tags --force --progress -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.066Z]  > /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:40.070Z]  > /export/home/anaconda/2020.02/bin/git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.086Z]  > /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:40.102Z]  > /export/home/anaconda/2020.02/bin/git show-ref --tags -d # timeout=10
[2022-04-27T20:26:40.118Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:40.118Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:40.119Z]  > /export/home/anaconda/2020.02/bin/git fetch --tags --force --progress --prune -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:40.666Z] Checking out Revision 7912da59315e5959de99fc58a190171d74f3198b (master)
[2022-04-27T20:26:40.706Z] Commit message: "retry double checkout"
[2022-04-27T20:26:40.672Z]  > /export/home/anaconda/2020.02/bin/git config core.sparsecheckout # timeout=10
[2022-04-27T20:26:40.676Z]  > /export/home/anaconda/2020.02/bin/git checkout -f 7912da59315e5959de99fc58a190171d74f3198b # timeout=10
[2022-04-27T20:26:40.712Z]  > /export/home/anaconda/2020.02/bin/git rev-list --no-walk b065c52da9fb8e0a07ca2e0d3dba544438412bf6 # timeout=10
[2022-04-27T20:26:41.332Z] [GitHub Checks] GitHub check (name: Jenkins, status: in_progress) has been published.
[2022-04-27T20:26:42.008Z] [Pipeline] checkout
[2022-04-27T20:26:42.023Z] The recommended git tool is: NONE
[2022-04-27T20:26:42.193Z] using credential github-app
[2022-04-27T20:26:42.211Z] Fetching changes from the remote Git repository
[2022-04-27T20:26:42.229Z] Pruning obsolete local branches
[2022-04-27T20:26:42.229Z] Pruning obsolete local tags that do not exist on remotes
[2022-04-27T20:26:42.524Z] Fetching without tags
[2022-04-27T20:26:42.832Z] Checking out Revision 7912da59315e5959de99fc58a190171d74f3198b (master)
[2022-04-27T20:26:42.864Z] Commit message: "retry double checkout"
[2022-04-27T20:26:42.203Z]  > /export/home/anaconda/2020.02/bin/git rev-parse --resolve-git-dir /export/data/ci/jenkins/workspace/s_jenkins-git-no-tags_master/.git # timeout=10
[2022-04-27T20:26:42.219Z]  > /export/home/anaconda/2020.02/bin/git config remote.origin.url https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:42.235Z]  > /export/home/anaconda/2020.02/bin/git show-ref --tags -d # timeout=10
[2022-04-27T20:26:42.253Z]  > /export/home/anaconda/2020.02/bin/git --version # timeout=10
[2022-04-27T20:26:42.256Z]  > git --version # 'git version 2.35.3'
[2022-04-27T20:26:42.256Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:42.257Z]  > /export/home/anaconda/2020.02/bin/git ls-remote -t -- https://github.com/ilCatania/jenkins-git-no-tags.git # timeout=10
[2022-04-27T20:26:42.531Z] Fetching upstream changes from https://github.com/ilCatania/jenkins-git-no-tags.git
[2022-04-27T20:26:42.531Z] using GIT_ASKPASS to set credentials GitHub integration for ilCatania
[2022-04-27T20:26:42.531Z]  > /export/home/anaconda/2020.02/bin/git fetch --tags --force --progress --prune -- https://github.com/ilCatania/jenkins-git-no-tags.git +refs/heads/master:refs/remotes/origin/master # timeout=10
[2022-04-27T20:26:42.839Z]  > /export/home/anaconda/2020.02/bin/git config core.sparsecheckout # timeout=10
[2022-04-27T20:26:42.842Z]  > /export/home/anaconda/2020.02/bin/git checkout -f 7912da59315e5959de99fc58a190171d74f3198b # timeout=10
[2022-04-27T20:26:43.545Z] [GitHub Checks] GitHub check (name: Jenkins, status: in_progress) has been published.

```

<p>I guess my main point is, if someone is trying to set this up, he's in for a lot of weirdness and several hours trying to figure it out</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/saper.png" title="saper's avatar" /> saper:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67984?focusedId=425186&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-425186">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Maybe to would be more helpful if your Jenkinsfile tried fetching only once and then try to check if any tags are available indeed (I have added 1.0 in my fork - <a href="https://github.com/saper/jenkins-git-no-tags/releases/tag/1.0" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/saper/jenkins-git-no-tags/releases/tag/1.0</a> - maybe just a simple <tt>git tag -l</tt> check</p>

<p>Also your output is a bit different now - can you try to update your plugins and maybe Jenkins - since I could not reproduce your problem at all using newest plugins.</p>
</pre>
</details>

</ul>
</details>

<p>Maybe to would be more helpful if your Jenkinsfile tried fetching only once and then try to check if any tags are available indeed (I have added 1.0 in my fork - <a href="https://github.com/saper/jenkins-git-no-tags/releases/tag/1.0" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/saper/jenkins-git-no-tags/releases/tag/1.0</a> - maybe just a simple <tt>git tag -l</tt> check</p>

<p>Also your output is a bit different now - can you try to update your plugins and maybe Jenkins - since I could not reproduce your problem at all using newest plugins.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-47379%22">JENKINS-47379</a>]</i>
